app/src/main/java/pl/droidevs/books/library/LibraryAdapter.java

```diff
@@ -73,7 +72,7 @@ public void addItem(Book book, int position) {
 
     @FunctionalInterface
     public interface BookItemClickListener {
-        void onBookClicked(@NonNull View view, @NonNull String bookId, @NonNull Integer index);
+        void onBookClicked(@NonNull View view, @NonNull BookId bookId, @NonNull Integer index);
```

Comment:
generalnie moim zdaniem powinniśmy używać całego obiektu BookId a nie stringa w nim zawartego. Jako, że jest to obiekt serializowalny to można przekazywać przez intenta całe BookId.